src/lib/validateRoute.tsx

```diff
@@ -0,0 +1,127 @@
+import React, {
+  createContext,
+  useContext,
+  ReactNode,
+  useState,
+  useEffect,
+} from 'react'
+import { usePathname, useRouter } from 'next/navigation'
+import { useAppSelector } from '@/store/hooks'
+import { matchRoute } from './matchRoute'
+import { getUserPermissions } from './getUserPermissions'
+import { selectToken } from '@/modules/auth/authSlice'
+import { protectedRoutes, publicRoutes } from '@/configs/routes'
+import { LoadingSpinner } from '@/components/ui/loadingSpinner'
+
+interface RouteContextType {
+  isPublic: boolean
+  isProtected: boolean
+  currentRoute: string | null
+  requiredPermission: string | null
+  hasPermission: boolean
+}
+
+const RouteContext = createContext<RouteContextType | undefined>(undefined)
+
+export const ValidateRoute = ({
+  children,
+}: {
+  children: ReactNode
+}): React.JSX.Element => {
+  const router = useRouter()
+  const pathname = usePathname()
+
+  const token = useAppSelector(selectToken)
+  const [loading, setLoading] = useState(true)
+
+  const publicMatch = matchRoute(pathname, publicRoutes)
+  const isPublic = Boolean(publicMatch)
+
```

Comment:
burada noluyo? biz public routeları zaten hardcoded bi şekilde yazmışız neden matchroute ile check ediyoruz public mi diye?

Reply:
protected'ın yanında public routesde token'ı olan public giriş yapılan adresler olduğu için onuda kontrol etmek gerekiyor. isPublic gerekiyor gibi geliyor şuan bana bakıyorum farklı fikirler.

Reply:
"token'ı olan public giriş yapılan adresler" tokeni olan adresler derken ne demek istedin?

benim token diye bahsettiğim hep kullanıcının tokenı. bir adres ya publictir ve token'a hiç ihtiyacı yoktur, ya da protected idir ve tokensız girilmez diye düşünüyorum. hem public hem de protected olan bir route var mı? ya da şöyle sorayım isProtected ile isPublic'in birbirine zıt olmadığı bir ihtimal var mı?

Reply:
hiç sorun yok, ama tekrar request istemeden önce konuşalım gerekli mi gereksiz mi, düzeltme lazım mı değil mi tartışıp karar verirsek ondan sonrasında request istersen zamandan kazanırız diye düşünüyorum